AK/Tuple.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/StdLibExtras.h>
+#include <AK/TypeList.h>
+
+namespace AK::Detail {
+
+template<typename... Ts>
+struct Tuple {
+};
+
+template<typename T>
+struct Tuple<T> {
```

Comment:
I think this should be a class. Same with the other structs in this PR.

Reply:
I could do that, there's no particular difference in this application
alternatively I could make its members private.

Reply:
I personally like structs with private bits for metaprogramming applications, one `public:` is easier to type than two `private:`'s

Reply:
I made the members private, so it should have the same effect as a class
To mirror most other metaprogramming utils, I'm gonna keep it a struct though...

Reply:
Yeah I was just going by the coding guidelines, but there is already precedence for using structs in some cases so I guess it's ok.